### PR TITLE
PlainTextBenchmark refactor

### PR DESCRIPTION
**Pull request by ShrutiVerma97**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 1]
```diff
@@ -720,6 +726,8 @@ object Http {
     def apply[R, E, B](f: A => HExit[R, E, B]): Http[R, E, A, B] = FromFunctionHExit(f)
   }
 
+  private final case class When[R, E, A, B](f: A => Boolean, other: Http[R, E, A, B]) extends Http[R, E, A, B]
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/Http.scala]: We can define `IfThenElse` instead and can derive when from it. Let's discuss it.
cc: @tusharmath

File: zio-http/src/main/scala/zhttp/http/Http.scala [hunk 2]
```diff
@@ -77,6 +77,11 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
    */
   final def apply(a: A): ZIO[R, Option[E], B] = execute(a).toZIO
 
+  /**
+   * Applies Http based only if the condition function evaluates to true
+   */
+  final def when[R1 <: R, E1 >: E, A2 <: A, B1 >: B](f: A2 => Boolean): Http[R, E, A2, B] = Http.When(f, self)
+
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/http/Http.scala:90]: Add HttpApp specific operators like `whenPath`, `whenHeaders` and `whenRequest`

File: example/src/main/scala/example/PlainTextBenchmarkServer.scala [hunk 3]
```diff
@@ -21,15 +21,17 @@ object Main extends App {
     .withServer(STATIC_SERVER_NAME)
     .freeze
 
+  val path = "/plaintext"
+
+  def app(res: Response) = Http.succeed(res).when((a: Request) => a.url.path.toString.equals(path))
```
Comment (amitksingh1490) [inline example/src/main/scala/example/PlainTextBenchmarkServer.scala]: use `whenPath` instead

**Review by amitksingh1490 — APPROVED** [review]

File: example/src/main/scala/example/PlainTextBenchmarkServer.scala (hunk 3 above)
Comment (amitksingh1490) [reply example/src/main/scala/example/PlainTextBenchmarkServer.scala]: as per below comment

**Review by amitksingh1490 — CHANGES_REQUESTED** [review]

File: zio-http/src/main/scala/zhttp/http/Http.scala (hunk 2 above)
Comment (tusharmath) [reply zio-http/src/main/scala/zhttp/http/Http.scala:90]: We aren't using all the type params R1, E1 etc., why have them?

Comment (tusharmath) [reply zio-http/src/main/scala/zhttp/http/Http.scala:90]: We should have two overloads for `whenPath` one should take a `Path` and one should take a `String`.

File: zio-http/src/main/scala/zhttp/http/Http.scala (hunk 1 above)
Comment (tusharmath) [reply zio-http/src/main/scala/zhttp/http/Http.scala]: We can use `Http.fromFunction[A]` instead I think.

**Pull request closed without merging**